doc/sources/oneapi_gpu.rst

```diff
@@ -0,0 +1,107 @@
+.. * you may not use this file except in compliance with the License.
+.. * You may obtain a copy of the License at
+.. *
+.. *     http://www.apache.org/licenses/LICENSE-2.0
+.. *
+.. * Unless required by applicable law or agreed to in writing, software
+.. * distributed under the License is distributed on an "AS IS" BASIS,
+.. * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+.. * See the License for the specific language governing permissions and
+.. * limitations under the License.
+.. *******************************************************************************/
+
+.. _oneapi_gpu:
+
+##############################################################
+oneAPI and GPU support in Intel(R) Extension for Scikit-learn*
+##############################################################
+
+Intel(R) Extension for Scikit-learn* supports oneAPI concepts, which
+means that algorithms can be executed on different devices: CPUs and GPUs.
+This is done via integration with
+`dpctl <https://intelpython.github.io/dpctl/latest/index.html>`_ package that
+implements core oneAPI concepts like queues and devices.
+
+Prerequisites
+-------------
+
+For execution on GPU, DPC++ compiler runtime and driver are required. Refer to `DPC++ system
+requirements <https://software.intel.com/content/www/us/en/develop/articles/intel-oneapi-dpcpp-system-requirements.html>`_ for details.
+
+DPC++ compiler runtime can be installed either from PyPI or Anaconda:
+
+- Install from PyPI::
+
+     pip install dpcpp-cpp-rt
+
+- Install from Anaconda::
+
+     conda install dpcpp_cpp_rt -c intel
+
+Device offloading
+-----------------
+
+Intel(R) Extension for Scikit-learn* offers two options for running an algorithm on a
+specific device with the help of dpctl:
+
+1. Pass input data as `dpctl.tensor.usm_ndarray <https://intelpython.github.io/dpctl/latest/docfiles/dpctl.tensor_api.html#dpctl.tensor.usm_ndarray>`_ to the algorithm.
+
+   The computation will run on the device where the input data is
+   located, and the result will be returned as :code:`usm_ndarray` to the same
+   device.
+
+   .. note::
+     All the input data for an algorithm must reside on the same device.
+
+   .. warning::
+     The :code:`usm_ndarray` can only be consumed by the base methods
+     like :code:`fit`, :code:`predict`, and :code:`transform`.
+     Note that only the algorithms in Intel(R) Extension for Scikit-learn* support
+     :code:`usm_ndarray`. The algorithms from the stock version of scikit-learn
+     do not support this feature.
+2. Use global configurations of Intel(R) Extension for Scikit-learn\*:
+     1. The :code:`target_offload` option can be used to set the device primarily
+        used to perform computations. Accepted data types are :code:`str` and
+        :code:`dpctl.SyclQueue`. If string, expected to be "auto" (the execution
+        context is deduced from input data location), or SYCL* filter selector
+        string. Default value is "auto"
```

Comment:
```suggestion
        :code:`dpctl.SyclQueue`. If you pass a string to :code:`target_offload`,
        it should either be ``"auto"``, which means that the execution
        context is deduced from the location of input data, or a string
        with SYCL* filter selector. The default value is ``"auto"``.
```

Reply:
@vlad-nazarov please take a look at this wording

Reply:
@outoftardis, this wording looks good to me